tools/tests_wycheproof_generate.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 Random "Randy" Lattice and Sean Andersen
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+'''
+Generate a C file with ECDSA testvectors from the Wycheproof project.
+'''
+
+import json
+import hashlib
+import urllib.request
+import sys
+
+filename_input = sys.argv[1]
+
+with open(filename_input) as f:
+    doc = json.load(f)
+
+num_groups = len(doc['testGroups'])
+
+def to_c_array(x):
+    if x == "": return ""
+    s = ',0x'.join(a+b for a,b in zip(x[::2], x[1::2]))
+    return "0x" + s
+
+
+num_vectors, max_msg_size, max_sig_size = 0, 0, 0
+offset_messages, offset_signatures = 0, 0
+out = ""
+messages = ""
+signatures = ""
+
+for i in range(num_groups):
+    group = doc['testGroups'][i]
+    num_tests = len(group['tests'])
+    public_key = group['publicKey']
+    for j in range(num_tests):
+        test_vector = group['tests'][j]
+        # // 2 to convert hex to byte length
+        sig_size = len(test_vector['sig']) // 2
+        msg_size = len(test_vector['msg']) // 2
+        max_sig_size = max(sig_size, max_sig_size)
+        max_msg_size = max(msg_size, max_msg_size)
+
+        if test_vector['result'] == "invalid": expected_verify = 0
+        elif test_vector['result'] == "valid": expected_verify = 1
+        else: raise ValueError("invalid result field")
+
+        if num_vectors != 0 and msg_size != 0: messages += ", "
+        if num_vectors != 0 and sig_size != 0: signatures += ", "
```

Comment:
```suggestion
        if num_vectors != 0 and msg_size != 0: messages += ",\n"
        if num_vectors != 0 and sig_size != 0: signatures += ",\n"
```
This will result in much more readable output. And we should avoid very long lines. Some tools don't like them, e.g., some editors become very slow.